include/seqan3/range/decorator/gap_decorator_anchor_set.hpp

```diff
@@ -83,34 +79,42 @@ class gap_decorator_anchor_set
         //!\brief Stores the virtual position index for the seqan3::gap_decorator_anchor_set.
         typename gap_decorator_anchor_set::size_type pos{0u};
         //!\brief Stores the physical position in the ungapped/underlying view.
-        int32_t ungapped_view_pos{0u};
+        int64_t ungapped_view_pos{0u};
         //!\brief Stores the position (inkl. gaps) where the last (consecutive) gap that is still before the current
         //!       iterator position ends.
         typename gap_decorator_anchor_set::size_type left_gap_end{0};
         //!\brief A pointer to the current anchor gap node. Note that the current tuple value at position 0 is the
         //!       start of the right gap that is still behind the current iterator position.
         typename gap_decorator_anchor_set::set_iterator_type anchor_set_it{};
+        //!\brief Cashes the current value pointed at.
+        typename gap_decorator_anchor_set::value_type current_val{gap{}};
 
         //!\brief A helper function that performs the random access into the anchor set, updating all member variables.
         void jump(typename gap_decorator_anchor_set::size_type const new_pos)
         {
-            assert(new_pos <= (*host).size());
+            assert(new_pos <= host->size());
             pos = new_pos;
 
-            anchor_set_it = (*host).anchors.upper_bound(anchor_gap_t{pos, (*host).bound_dummy});
+            anchor_set_it = host->anchors.upper_bound(anchor_gap_t{pos, host->bound_dummy});
             ungapped_view_pos = pos;
 
-            if (anchor_set_it != (*host).anchors.begin())
+            if (anchor_set_it != host->anchors.begin())
             {
                 typename gap_decorator_anchor_set::set_iterator_type prev{std::prev(anchor_set_it)};
                 size_type gap_len{(*prev).second};
 
-                if (prev != (*host).anchors.begin())
+                if (prev != host->anchors.begin())
                     gap_len -= (*(std::prev(prev))).second;
 
                 ungapped_view_pos -= (*prev).second;
                 left_gap_end = (*prev).first + gap_len;
             }
+
+            if (ungapped_view_pos != static_cast<int64_t>(host->ungapped_view.size()) &&
```

Comment:
if `ungapped_view_pos` is of type `gap_decorator_anchor_set::size_type` you don't need the `static_cast` anymore

Reply:
see above